tests/functional/raw-node/test/routes/routeBackbeat.js

```diff
@@ -452,4 +452,103 @@ describeSkipIfAWS('backbeat routes:', () => {
             });
         });
     });
+    describe('Batch Delete Route', () => {
+        it('should batch delete a location', done => {
+            let versionId;
+            let location;
+
+            async.series([
+                done => s3.putObject({
+                    Bucket: TEST_BUCKET,
+                    Key: 'batch-delete-test-key',
+                    Body: new Buffer('hello'),
+                }, done),
+                done => s3.getObject({
+                    Bucket: TEST_BUCKET,
+                    Key: 'batch-delete-test-key',
+                }, (err, data) => {
+                    assert.ifError(err);
+                    assert.strictEqual(data.Body.toString(), 'hello');
+                    versionId = data.VersionId;
+                    done();
+                }),
+                done => {
+                    makeBackbeatRequest({
+                        method: 'GET', bucket: TEST_BUCKET,
+                        objectKey: 'batch-delete-test-key',
+                        resourceType: 'metadata',
+                        authCredentials: backbeatAuthCredentials,
+                        queryObj: {
+                            versionId,
+                        },
+                    }, (err, data) => {
+                        assert.ifError(err);
+                        assert.strictEqual(data.statusCode, 200);
+                        const metadata = JSON.parse(
+                            JSON.parse(data.body).Body);
+                        location = metadata.location;
+                        done();
+                    });
+                },
+                done => {
+                    const options = {
+                        authCredentials: backbeatAuthCredentials,
+                        hostname: ipAddress,
+                        port: 8000,
+                        method: 'POST',
+                        path: '/_/backbeat/batchdelete',
+                        requestBody:
+                        `{"Locations":${JSON.stringify(location)}}`,
+                        jsonResponse: true,
+                    };
+                    makeRequest(options, done);
+                },
+                done => s3.getObject({
+                    Bucket: TEST_BUCKET,
+                    Key: 'batch-delete-test-key',
+                }, err => {
+                    // should error out as location shall no longer exist
+                    assert(err);
```

Comment:
@bennettbuchanan I reverted the check of `NoSuchKey` because what's missing is the data location, not the object itself, so it ends up erroring with a server error (503), and I don't think we need to be specific about this error.